app.py

```diff
@@ -0,0 +1,411 @@
+import copy
+import datetime
+import json
+import os
+import time
+
+from flask import Flask, abort, render_template, request, session, make_response
+from google.auth.transport import requests
+from google.cloud import datastore, storage, secretmanager
+import google.oauth2.id_token
+
+from cron.photos import hash_username
+
+app = Flask(__name__)
+
+verify_firebase_token = None
+datastore_client = None
+SCHEDULE_INFO = None
+DAYS = None
+FALL_TRI_END = datetime.datetime(2020, 11, 23, 15, 30, 0, 0)
+WINT_TRI_END = datetime.datetime(2021, 3, 6, 15, 30, 0, 0)
+
+def init_app(test_config=None):
+    global verify_firebase_token
+    global datastore_client
+    global SCHEDULE_INFO
+    global DAYS
+    app.permanent_session_lifetime = datetime.timedelta(days=3650)
+    if test_config is None:
+        # Authenticate ourselves
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "service_account.json"
+
+        # Get application secret key
+        secret_client = secretmanager.SecretManagerServiceClient()
+        app.secret_key = secret_client.access_secret_version(request={
+            "name": "projects/epschedule-v2/secrets/session_key/versions/1"
+        }).payload.data
+
+        verify_firebase_token = lambda token: \
+            google.oauth2.id_token.verify_firebase_token(
+                token, requests.Request())
+
+        storage_client = storage.Client()
+        data_bucket = storage_client.bucket("epschedule-data")    
+        SCHEDULE_INFO = json.loads(
+            data_bucket.blob("schedules.json").download_as_string())
+        DAYS = json.loads(
+            data_bucket.blob("master_schedule.json").download_as_string())
+
+        datastore_client = datastore.Client()
+    else:
```

Comment:
Why are we manually passing fake data in with our own hacky function? It'd be much cleaner to just `@patch` away the Google Cloud libraries, which also means we don't have to modify `main.py` at all (and can remove `app.py` entirely).